### PR TITLE
[ci] Move continueOnError flag for Bazel fetch step

### PR DESCRIPTION
**Pull request by milesdai**
This is an attempt to try and trigger a warning on Github instead of a failure. Although Azure registers an error in this job as a warning, it seems to show up on Github as a failure.

Maybe moving the continueOnError flag to a step instead of the entire job (as we do in the flake8 linting step) might help.

Signed-off-by: Miles Dai <milesdai@google.com>

### CODE REVIEW

**Review by drewmacrae — APPROVED** [review]

File: azure-pipelines.yml
Comment (drewmacrae) [reply azure-pipelines.yml:121]: It's the result of the previous step, job or stage which should prevent this step from trying to publish a tarball that isn't present

**Review by marnovandermaas — APPROVED** [review]
LGTM, thanks

**Comment by a-will** [comment]
After reading some docs, I think github only has a binary success/failed status for completed checks, but runs can be annotated to get warnings. It appears the `task.logissue` command can be used to create those warning messages. Perhaps an always-successful job + check run annotations is the way to get the warnings you're looking for?

**Comment by milesdai** [comment]
A quick summary of what's going on in this PR:

The `Prefetch Bazel deps` job has two steps:
1. Download and package the dependencies
2. Publish the dependencies as an artifact for later stages to download

The problem we were facing was that the download step is flaky, so we don't want it to stall the rest of the pipeline. Azure has a flag called `continueOnTrue` that we can add to this job that will allow the pipeline to continue if this job fails. Unfortunately, applying this flag to the entire job causes the job to show up as failed on Github which would require committers to potentially ignore failing CI steps before merging. Instead, we want Github to silently ignore the error since this step is optional.

To solve this, we add the `continueOnError` flag to step 1 specifically (instead of the job as a whole), and we add a condition to step 2 to only run if step 1 succeeded (otherwise step 2 will fail and cause the whole job to fail as well). The subtlety here is that using `condition: succeeded()` does not quite work because, as described in the [documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeeded), `succeeded` passes if the `Agent.JobStatus` pipeline variable equals *either* `Succeeded` or `SucceededWithIssues`. Adding the `continueOnError` flag to step 1 causes the `JobStatus` to become `SucceededWithIssues` rather than `Failed` which means step 2 would run even if step 1 fails. Therefore, we need the stricter and more verbose `condition: eq(variables['Agent.JobStatus'], 'Succeeded')` to prevent step 2 from running.

Thanks @a-will for helping with the debugging.

**Review by a-will — APPROVED** [review]

**Pull request merged**
